File: python/problem_002.py
'''
'''
# Answer: 4613732

def fibonacci(n):
    (i, tmp, fib1, fib2) = (0, 0, 0, 1)
    while i < n:
        i += 1
        tmp = fib1
        fib1 = fib2
        fib2 = tmp + fib2
    return fib2

def fibonacci_lesseq(n):
    fibs = []
    i = 0
    while fibonacci(i) <= n:
        fibs.append(fibonacci(i))
        i += 1
    return fibs

def is_even(n):
    return 0 == n % 2

answer_002 = sum(filter(is_even, fibonacci_lesseq(4000000)))

print("Answer: " + str(answer_002) + "\n")

File: c/problem_002.c
// Problem 2

#include <stdio.h>

int fibonacci(int n) {
  int fib1 = 0;
  int fib2 = 1;
  int tmp = 0;

  while (n > 0) {
    n--;
    tmp = fib1;
    fib1 = fib2;
    fib2 += tmp;
  }

  return fib1;
}

int main() {
  int answer_002 = 0;
  int i = 0;
  int fib = fibonacci(i);

  while (fib <= 4000000) {
    if (fib % 2 == 0)
      answer_002 += fib;
    fib = fibonacci(i);
    i++;
  }

  printf("SOLUTION: %i\n", answer_002);
}

// SOLUTION: 4613732
